### DISCUSSION TITLE
Compile for P3D V5

### DISCUSSION THREAD

**Discussion by Killerbee1958** [nguyenquyhy/Flight-Tracker-StreamDeck#198] [General]
I am impressed with the working of Flight Tracker for MSFS however i am still mainly flying in P3D V5
Now i noticed a remark on the intro page:

The current code target x64 and the latest SimConnect from Microsoft Flight Simulator. However, if you compile for x86 and reference SimConnect SDK from FSX, this plugin should also work with FSX and P3D.

I have tried to use the flight tracker with P3D V5 but the passing of simconnect variables to P3D does not work. Passing a hotkey works btw if i start the elagato streamdeck in adminstrator mode.
Two questions
1. As far as i know P3D V5 is also X64 and has a working simconnect (i know as my Active Sky connects with P3D through simconnect).  So why is it not working with P3D V5?
2.  If the answer is that part of P3D is still X86 and i should recompile flight tracker, then can someone tell me how?  (Sorry, i am in IT but not a programmer)  :-)

**Comment by nguyenquyhy** [comment]
It's mainly due to the way SimConnect handle forward/backward compatibility. If the tool uses older SimConnect, it can work with newer Sim, but if the tool uses newer SimConnect, it won't be able to find SimConnect at all because the sim doesn't provide the version. 

So to get this to work with P3D, there are 3 ways:
- Change the tool back to use FSX's SimConnect. This allows the tool to work with FSX, P3D and MSFS at the same time, but that also means giving up anything that is unique to MSFS's SimConnect. I think there is not much that is added to SimConnect in MSFS right now, but I haven't really keeping track. The bigger problem is that the tool (or part of the tool) has to go back to 32-bit, which is a pain to support.
- Change the tool to use P3D's SimConnect. P3D is already 64-bit, so it's fine on this front. But there are some API differences in P3D and MSFS from the base FSX.
- Use [CFG file](https://docs.flightsimulator.com/html/Programming_Tools/SimConnect/SimConnect_CFG_Definition.htm) to change it to using TCP. Since SimConnect is designed to work over network via TCP, you can set this to instruct it to point back to localhost (127.0.0.1). This eliminates the direct coupling between the tool and the sim, so dll version number no longer matter.

If you want to do the 1st 2 ways, you have to first install SimConnect SDK (e.g. inside Prepar3D SDK), then change FlightStreamDeck.SimConnectFSX.csproj to point to that dll instead of Microsoft.FlightSimulator.SimConnect.dll, and recompile the solution FlightStreamDeck.sln.

**Reply by Killerbee1958** [reply]
As i do not know how to compile flighttracker i guess only option3 is available to me.   :-)
Where should i put the simconnect.cfg file to make this happen?

**Reply by nguyenquyhy** [reply]
Using the CFG is actually the suggested solution in MSFS SDK for backward compatibility. I don't have time to try that yet so I don't have much idea, but pretty sure you can find the info on the internet. The file is there since FSX.

**Reply by Killerbee1958** [reply]
Where is the file FlightStreamDeck.SimConnectFSX.csproj located on my harddrive?

**Reply by nguyenquyhy** [reply]
It's in the source code.

**Reply by nguyenquyhy** [reply]
Here is a thread about network configuration https://forums.flightsimulator.com/t/simconnect-on-2nd-pc-network-setup/365647
